File: src/codility/java/counting/PermCheck.java
package codility.java.counting;

import java.util.HashSet;

public class PermCheck {
    public int solution(int[] A) {
        HashSet<Integer> set = new HashSet<>();
        for (int a : A) {
            set.add(a);
        }

        for (int i = 1;i<=A.length;i++) {
            if (!set.contains(i)) {
                return 0;
            }
        }
        return 1;
    }
}

File: src/codility/kotlin/PermCheck.kt
package codility.kotlin

class PermCheck {

    fun solution(A: IntArray): Int {
        val set = A.toHashSet()

        for (i in 1..A.size) {
            if (!set.contains(i)) {
                return 0
            }

        }
        return 1
    }

}